Create team dashboard template

geodn opened the pull request:
Working version: https://web.airstage.app/views/vew20220914z6hfn588ya3?__env=prod

We display the unique products per week, unique customers per week, top products, and orders per week. We also allow the the user to select a specific customer and show data specific to them.

rose-huang commented on team_dashboard/team_dashboard/team_dashboard.view.tsx:
```diff
@@ -0,0 +1,65 @@
+        />
+        <Table title="Orders per week" task="get_orders_per_week" />
+      </Stack>
+      <Select
```
Can we add some titles or text in the template to distinguish between the top tables (which seem like an overview of all the external facing stuff) and the bottom that is an overview per customer? I think this would make the template a bit more clear. Or maybe you can have the orders per week and top products show for all of the customers if no customer is selected and then filter on specific customer if you select someone in the dropdown?

rose-huang commented on team_dashboard/team_dashboard/team_dashboard.view.tsx:
```diff
@@ -0,0 +1,65 @@
+        <Table
+          title="Top products"
+          task="list_top_products"
+          columns={[{ accessor: "cnt", label: "num_orders" }]}
```
Let's make the labels more human readable in the title case format of "Column name" ... so something like "Count" or "Number of orders". same for all other columns

rose-huang commented on team_dashboard/README.md:
```diff
@@ -0,0 +1,9 @@
+
+- Navigate to the team_dashboard directory: `cd team_dashboard`
+- Deploy tasks: `airplane deploy tasks --yes`
+- Develop your view locally: `airplane views dev`
```
Can you mention what demo resources to replace before or after this step?  

I think Zhan has good instructions here https://github.com/airplanedev/templates/blob/main/database_issues_alert/README.md (although should prob add a step for airplane dev in his)

geodn replied:
Done.

rose-huang reviewed (changes requested):
TY so much for working on the template!

geodn commented on team_dashboard/team_dashboard/team_dashboard.view.tsx:
```diff
@@ -0,0 +1,65 @@
+  return (
+    <Stack spacing="md">
+      <Stack direction="row" spacing="xl" grow>
+        <Table title="Employees" task="list_employees" />
```
Removed.

geodn commented on team_dashboard/team_dashboard/team_dashboard.view.tsx:
```diff
@@ -0,0 +1,65 @@
+} from "@airplane/views";
+
+// Views documentation: https://docs.airplane.dev/views/getting-started
+const TeamDashboard = () => {
```
Renamed to customer_insights

geodn commented:
Updated the link to the dashboard, since it got renamed. I added a new table, unique products per week, to fill the hole left by deleting the employees table.

rose-huang commented on customer_insights/README.md:
```diff
@@ -0,0 +1,10 @@
+## Next steps
+
+- Navigate to the customer_insights directory: `cd customer_insights`
+- If you plan on using this on a different database resource than the demo database, change the demo_db resource in customer_insights.task.yaml to your own database resource's slug
```
What is `customer_insights.task.yaml`? Can you be more specific with saying to replace the "[Demo DB]"  resource? It also should be replaced with the name of the resource, not slug, I believe.

rose-huang replied:
https://github.com/airplanedev/templates/tree/main/stripe_billing_dashboard is an example of telling users to make changes in multiple files

geodn replied:
Done.

rose-huang commented on customer_insights/customer_insights/customer_insights.view.tsx:
```diff
@@ -0,0 +1,91 @@
+          task="get_orders_per_week"
+          columns={[
+            { accessor: "cnt", label: "count" },
+            { accessor: "week", type: "date" },
```
please add a label to this! Also the labels should also be title case ie "Number of orders"

geodn replied:
Done.

rose-huang commented on customer_insights/tasks/get_customers_per_week.task.yaml:
```diff
@@ -0,0 +1,9 @@
+name: Get customers per week
+description: Get the number of unique customers in each week
+
+sql:
```
for all of the task yaml files, can you add a comment above the [Demo DB] resource that's like # Replace [Demo DB] below with your own database resource

geodn replied:
Done.

rose-huang reviewed (changes requested):
Thanks for making the changes! Could you update the link with the preview of the template in your description? 

Also sorry for all the nits - be sure to run through the guidance in this doc:  https://www.notion.so/Template-Gallery-how-to-contribute-53af3baef69d40f29a737c0c2522a7c7

rose-huang commented on customer_insights/README.md:
```diff
@@ -0,0 +1,10 @@
+# Template: Team Dashboard
```
nit: replace this with new name "Customer insights" in sentence case

geodn replied:
Done.

geodn commented on customer_insights/customer_insights/customer_insights.view.yaml:
```diff
@@ -0,0 +1,3 @@
+slug: customer_insights
```
Done.

rose-huang commented on customer_insights/README.md:
```diff
@@ -0,0 +1,10 @@
+# Template: Customer insights
```
this looks really great! Sorry one last nit: Can we rename this Template to "Customer insights dashboard" and the outer folder containing everything from `customer_insights` to `customer_insights_dashboard`? I think it will be less confusing if there isn't a nested folder with a view of the same name

rose-huang commented on customer_insights/README.md:
```diff
@@ -0,0 +1,10 @@
+
+## Next steps
+
+- Navigate to the customer_insights directory: `cd customer_insights`
```
can change this to cd `customer_insights_dashboard` as well after updating the folder name

geodn replied:
Done. Also updated package.json

rose-huang reviewed (approved):
Looks awesome, TY so much george 🙏
